src/backend/optimizer/path/allpaths.c

```diff
@@ -2125,15 +2125,38 @@ set_cte_pathlist(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
 			subplan = subquery_planner(cteroot->glob, subquery, cteroot, cte->cterecursive,
 									   tuple_fraction, &subroot, config);
 
-			cteplaninfo->shared_plan = prepare_plan_for_sharing(cteroot, subplan);
+			if (subplan->flow->locustype != CdbLocusType_SegmentGeneral)
+			{
```

Comment:
Can we omit the parentheses here and below?

Reply:
I've been confused a bit until guessed that you meant "braces"

Reply:
Yes, could you remove unnecessary braces?